test/azure/src/main/java/fixtures/azureparametergrouping/ParameterGroupings.java

```diff
@@ -78,7 +79,7 @@
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @return the {@link ServiceFuture&lt;Void&gt;} object.
      */
-    ServiceFuture<Void> postOptionalAsync(ServiceCallback<Void> serviceCallback);
+    ServiceFuture<Void> postOptionalAsync(@NonNull ServiceCallback<Void> serviceCallback);
```

Comment:
(I believe) we do actually accept null service callbacks in the case that someone just wants a future with no callback.

Reply:
https://github.com/Azure/autorest-clientruntime-for-java/blob/v2/client-runtime/src/main/java/com/microsoft/rest/v2/ServiceFuture.java#L41

Reply:
I may go through at some point and add NonNull to some of the runtime helper methods.